web/controllers/class_controller.ex

```diff
@@ -18,13 +19,20 @@ defmodule CoursePlanner.ClassController do
   def create(conn, %{"class" => class_params}) do
     changeset = Class.changeset(%Class{}, class_params, :create)
 
-    case Repo.insert(changeset) do
+    multi =
+      Multi.new
+      |> Multi.insert(:class, changeset)
```

Comment:
so my understanding was the transaction runs when someone triggers "Repo.transaction(multi)" . and multi has the list of all changes required.